### ISSUE TITLE
valid-jsdoc rule and return prefered form bug

### ISSUE THREAD

**Issue by Spy-Seth** [eslint/eslint#4205]
I want to use the `valid-jsdoc` rule to ensure the JSdoc `@return` tag are `@return` and not `@returns`. I've use this configuration: 

```
{
  "rules": {
    "valid-jsdoc": [2, {
        "prefer": { "return": "return" }, // Here the issue
        "requireParamDescription": false,
        "requireReturnDescription": false,
        "requireReturn": false
    }]
  }
}
```

But eslint trigger false positive error on all my files. It said: `Use @return instead` but I'm already using it.

Here a file example: 
 
```javascript
'use strict';

/**
 * @param {Number} productId
 * @param {Number} userId
 * @param {Number} quantity
 * @constructor
 */
var FooCommand = function FooCommand(productId, userId, quantity) {
    this.productId = productId;
    this.quantity = quantity;
    this.userId = userId;
};

/**
 * @return {Number} 
 */
FooCommand.prototype.getProductId = function () {
    return this.productId;
};

/**
 * @return {Number}
 */
FooCommand.prototype.getUserId = function () {
    return this.userId;
};

/**
 * @return {Number}
 */
FooCommand.prototype.getQuantity = function () {
    return this.quantity;
};

module.exports = FooCommand;
```

And the eslint result:

```
$ ./node_modules/eslint/bin/eslint.js FooCommand.js 

./FooCommand.js
  15:1  error  Use @return instead  valid-jsdoc
  22:1  error  Use @return instead  valid-jsdoc
  29:1  error  Use @return instead  valid-jsdoc

✖ 3 problems (3 errors, 0 warnings)
```

I other way, if I choose the `@returns` form in the configuration it's works perfectly.

I use the last eslint version (v1.7.2).

**Comment by nzakas** [comment]
Yeah, looks like an errant warning. We will take a look.

**Comment by nzakas** [comment]
Working on this.

**Comment by Spy-Seth** [comment]
:+1:  :beer:

**Comment by Spy-Seth** [comment]
It still have an issue with this rules:

```javascript
/**
 * @return {Number}
 */
function foo() {
    return 1;
};

/**
 * @returns {Number}
 */
function bar() {
    return 2;
};

/**
 * No return tag.
 */
function bar() {
    return 2;
};
```

```
{
  "rules": {
    "valid-jsdoc": [2, {
        "prefer": { "return": "return" }, 
        "requireParamDescription": false,
        "requireReturnDescription": false,
        "requireReturn": false
    }]
  }
}
```

Only one error is trigger with same config as above: `15:1   error    Missing JSDoc @returns for function  valid-jsdoc`. But the `@returns` tag line 9 should also trigger an error.
